vllm/model_executor/layers/fused_moe/deepep_ll_prepare_finalize.py

```diff
@@ -110,6 +113,28 @@ def max_num_tokens_per_rank(self) -> int | None:
     def topk_indices_dtype(self) -> torch.dtype | None:
         return torch.int64
 
+    def set_expert_routing_info(
+        self,
+        global_to_physical: torch.Tensor,
+        physical_to_global: torch.Tensor,
+        local_expert_global_ids: torch.Tensor,
+    ) -> None:
+        self.global_to_physical = global_to_physical
+        self.physical_to_global = physical_to_global
+        self.local_expert_global_ids = local_expert_global_ids
+
+    def _map_global_to_physical_ids(self, topk_ids: torch.Tensor) -> torch.Tensor:
+        if self.global_to_physical is None:
+            return topk_ids
+        physical = self.global_to_physical[topk_ids.to(torch.long)]
+        return physical.to(topk_ids.dtype)
```

Comment:
Are the casts necessary here?  `topk_ids` should already be an int of some type and if needed, `global_to_physical` could be cast in `set_expert_routing_info` (or `__init__`) using `topk_indices_dtype` so that there's no issues about type mismatches.